Frogies-Project/Assets/Scripts/WaveSystem/WaveController.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Enemies;
+using StatsSystem.Health;
+using UnityEngine;
+using UnityEngine.Video;
+using Object = UnityEngine.Object;
+
+namespace WaveSystem
+{
+    public class WaveController
+    {
+        public event Action OnWaveCleared;
+        public event Action OnWaveStarted;
+        public event Action OnEnemyKilled;
+
+        private Dictionary<GameObject, bool> _enemyDictionary;
+        private List<GameObject> _spawners;
+        private List<GameObject> _enemies;
+        private Dictionary<Wave, Wave> _availableWaves;
```

Comment:
I have a few issues with it:
1. Wave doesn't override equality operator(you can read about it).
2. I dont see sense of this dictionary. What does it try to achieve?